### PR TITLE
Feature/20: merge existing oss code to release branch

### PR DESCRIPTION
**Pull request by rongwang0930**
TODO: some metastor & bdsql hardcode github urls needs to be modified later on.

### CODE REVIEW

File: mdl/src/main/cft/mdlHerdRds.yml [hunk 1]
```diff
@@ -151,7 +151,7 @@ Resources:
         rds.force_ssl: '1'
   PostgresDB:
     Type: 'AWS::RDS::DBInstance'
-    DeletionPolicy: Snapshot
+    DeletionPolicy: Delete
```
Comment (kusid) [inline mdl/src/main/cft/mdlHerdRds.yml]: Why would you delete the final snapshot?

Comment (rongwang0930) [reply mdl/src/main/cft/mdlHerdRds.yml]: I was thinking to add this as a cleanup step which deletes SNAPSHOT after RDS deletion, if this is not an usual case, I can revert this back

Comment (kusid) [reply mdl/src/main/cft/mdlHerdRds.yml]: Yes please, revert this.

File: metastor/managedObjectLoader/src/main/workflow-def/singletonObjectsAddPartitionWorkflow.bpmn [hunk 2]
```diff
@@ -19,23 +19,23 @@
     <scriptTask id="prepareEmailContent" name="Prepare Email Content" scriptFormat="javascript" activiti:autoStoreVariables="false">
       <script>
       var emailMessage="";
-      if (createCluster_taskStatus == 'ERROR') {
```
Comment (kusid) [inline metastor/managedObjectLoader/src/main/workflow-def/singletonObjectsAddPartitionWorkflow.bpmn]: Has anything changed in this file?

File: metastor/metastoreOperations/samples/object-registration/objectRegistrationRequests/METASTORE-sample.txt [hunk 3]
```diff
@@ -22,4 +22,4 @@ namespace|businessObjectDefinitionName|businessObjectFormatUsage|businessObjectF
 # Sample Object registration request
 MDL|OBJ_1|PRC|TXT|DM Managed|add|
 MDL|OBJ_2|PRC|TXT|DM Managed|drop|
-MDL|OBJ_2|PRC|TXT|Singleton|add|
```
Comment (kusid) [inline metastor/metastoreOperations/samples/object-registration/objectRegistrationRequests/METASTORE-sample.txt]: ??

Comment (rongwang0930) [reply metastor/metastoreOperations/samples/object-registration/objectRegistrationRequests/METASTORE-sample.txt]: after I remove the additional sec lines you added, it's showing like this on git, not sure what happened here, could be extra ending line of file got removed

File: metastor/managedObjectLoader/src/main/workflow-def/singletonObjectsAddPartitionWorkflow.bpmn (hunk 2 above)
Comment (rongwang0930) [reply metastor/managedObjectLoader/src/main/workflow-def/singletonObjectsAddPartitionWorkflow.bpmn]: yeah, disregarding whitespace, there are some small changes, like ";"

File: bdsql/src/main/scripts/sql_auth.sh [hunk 4]
```diff
@@ -49,12 +49,12 @@ execute_cmd	"sudo start presto-server"
 
 mdlInstanceRowCount=0
 # Wait presto sever to be up
-while [ ${mdlInstanceRowCount} -ne 3 ] ; do
```
[the remark this answers is not in the document]
Comment (rongwang0930) [reply bdsql/src/main/scripts/sql_auth.sh]: not sure how this works before, but right now we get Access denied when running this query in master node without using credential.

File: mdl/src/main/cft/installMDL.yml [hunk 5]
```diff
@@ -32,7 +32,7 @@ Parameters:
     Type: String
   MDLInstanceName:
     Description: MDL Instance Name, lowercase letters and numbers with max length 15
-    AllowedPattern: '[a-z0-9]*'
+    AllowedPattern: '[a-z0-9_-]*'
```
[the remark this answers is not in the document]
Comment (rongwang0930) [reply mdl/src/main/cft/installMDL.yml]: i'll revert this to old one

File: mdl/src/main/cft/mdl.yml [hunk 6]
```diff
@@ -29,7 +29,7 @@ Parameters:
   MDLInstanceName:
     Description: MDL Instance Name, lowercase letters and numbers with max length 15
     Type: String
-    AllowedPattern: '[a-z0-9_]*'
+    AllowedPattern: '[a-z0-9_-]*'
     MaxLength: 15
```
[the remark this answers is not in the document]
Comment (rongwang0930) [reply mdl/src/main/cft/mdl.yml]: same

**Pull request merged**